train.py

```diff
@@ -27,10 +27,10 @@
 from model import Transformer, ModelArgs
 from torch.distributed import destroy_process_group, init_process_group
 from torch.nn.parallel import DistributedDataParallel as DDP
-
+import torch._dynamo
 from tinystories import Task
 from export import model_export
-
+torch._dynamo.config.suppress_errors = True
```

Comment:
not familiar with these, why did you add these two lines? are they safe? importing _modules feels wrong

Reply:
> not familiar with these, why did you add these two lines? are they safe? importing _modules feels wrong

Yes, this approach is extreme.
I just adjusted to turn off compilation optimization under MPS. Because the compilation of torch is not complete for MPS, it currently supports backend="aot_eager", but the speed is a little slower than not compiling.

Reply:
> not familiar with these, why did you add these two lines? are they safe? importing _modules feels wrong

In addition, the reason why adding these two lines can make mps run normally is because mps will report an error when compiling, and then ignore the compilation behavior. That is to say, mps training is not compiled.